allensdk/brain_observatory/behavior/data_objects/stimuli/templates.py

```diff
@@ -128,12 +128,31 @@ def to_nwb(self, nwbfile: NWBFile,
 
         nwbfile.add_stimulus_template(visual_stimulus_image_series)
 
-        # Add index for this template to NWB in-memory object:
+        if 'image_index' in stimulus_presentations.value:
+            nwbfile = self._add_image_index_to_nwb(
+                nwbfile=nwbfile, presentations=stimulus_presentations)
+
+        return nwbfile
+
+    def _add_image_index_to_nwb(
```

Comment:
Did you want to make this a static method?

Reply:
I didn't make it static because it accesses `self.value` which is the template data.